Build windows 64bit binary using UCRT64

wader opened the pull request:
Is the default and recommended msystem setting. Will produce binaries that are compatible with windows 10 and later.

Also run tests for 32bit build.

Related to #2831

wader committed:
Build windows 64bit binary using UCRT64

Is the default and recommended msystem setting. Will produce
binaries that are compatible with windows 10 and later.

Also run tests for 32bit build.

Related to #2831

nicowilliams commented:
LGTM!

wader commented:
Note wait with merge, i cherry-picked https://github.com/jqlang/jq/pull/2835 ontop to try

wader commented on .github/workflows/ci.yml:
```diff
@@ -196,19 +196,16 @@ jobs:
-            CC: x86_64-pc-msys
-            toolchain: gcc
-            msystem: msys
+            msystem: UCRT64
```
As we only ran windows tests for amd64 before i suspect the test failures might be some (new line character?) difference between msystem `MSYS` and `UCRT64`/`MINGW32`.

If new line issue we could use `-b` somehow but only on windows somehow? mess

nicowilliams committed:
Use jq -b in tests/shtest

nicowilliams committed:
Add Windows strptime

nicowilliams committed:
Make Windows-optional tests not run on Windows again

nicowilliams commented:
I think maybe the only thing to do is drop the question mark in one of the commits I added.

@wader don't forget not to clobber those comments, if you feel the need to push again.

wader commented:
> I think maybe the only thing to do is drop the question mark in one of the commits I added.

Fixed

> @wader don't forget not to clobber those comments, if you feel the need to push again.

Not sure i follow, you mean the diff/od -c stuff? that i removed and fixedup into the first commit now

nicowilliams commented:
> Not sure i follow, you mean the diff/od -c stuff? that i removed and fixedup into the first commit now

I meant "commits", but I was on my phone and the swyping came off as "comments" :)

wader commented:
> I meant "commits", but I was on my phone and the swyping came off as "comments" :)

Suspected that, no worries :)
